cpp/models/ide_secir/parameters_io.cpp

```diff
@@ -61,161 +63,190 @@ IOResult<void> set_initial_flows(Model& model, ScalarType dt, std::string const&
-        model.m_transitions = TimeSeries<ScalarType>((int)InfectionTransition::Count);
+        model.m_transitions = TimeSeries<ScalarType>(Eigen::Index(InfectionTransition::Count) * num_age_groups);
     }
     if (model.m_populations.get_time(0) != 0) {
         model.m_populations.remove_last_time_point();
         model.m_populations.add_time_point<Eigen::VectorXd>(
-            0, TimeSeries<ScalarType>::Vector::Constant((int)InfectionState::Count, 0));
+            0, TimeSeries<ScalarType>::Vector::Constant((int)InfectionState ::Count * num_age_groups, 0));
     }
 
     // The first time we need is -4 * global_support_max.
     Eigen::Index start_shift = 4 * global_support_max_index;
     // The last time needed is dependent on the mean stay time in the Exposed compartment and
     // the mean stay time of asymptomatic individuals in InfectedNoSymptoms.
-    ScalarType mean_ExposedToInfectedNoSymptoms =
-        model.parameters.get<TransitionDistributions>()[Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)]
-            .get_mean(dt);
-    ScalarType mean_InfectedNoSymptomsToInfectedSymptoms =
-        model.parameters
-            .get<TransitionDistributions>()[Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms)]
-            .get_mean(dt);
-    Eigen::Index last_time_index_needed =
-        Eigen::Index(std::ceil((mean_ExposedToInfectedNoSymptoms + mean_InfectedNoSymptomsToInfectedSymptoms) / dt));
-    // Create TimeSeries with zeros. The index of time zero is start_shift.
-    for (Eigen::Index i = -start_shift; i <= last_time_index_needed; i++) {
-        // Add time point.
-        model.m_transitions.add_time_point(
-            i * dt, TimeSeries<ScalarType>::Vector::Constant((int)InfectionTransition::Count, 0.));
+    // The mean stay Time in a compartment may be dependent of the Age Group.
+    std::vector<ScalarType> mean_ExposedToInfectedNoSymptoms          = std::vector<ScalarType>(num_age_groups, 0.);
+    std::vector<ScalarType> mean_InfectedNoSymptomsToInfectedSymptoms = std::vector<ScalarType>(num_age_groups, 0.);
+    for (int group = 0; group < num_age_groups; group++) {
+        mean_ExposedToInfectedNoSymptoms[group] =
+            model.parameters
+                .get<TransitionDistributions>()[AgeGroup(group)]
+                                               [Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)]
+                .get_mean(dt);
+        mean_InfectedNoSymptomsToInfectedSymptoms[group] =
+            model.parameters
+                .get<TransitionDistributions>()[AgeGroup(group)]
+                                               [Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms)]
+                .get_mean(dt);
+        Eigen::Index last_time_index_needed = Eigen::Index(std::ceil(
+            (mean_ExposedToInfectedNoSymptoms[group] + mean_InfectedNoSymptomsToInfectedSymptoms[group]) / dt));
```

Comment:
With this we are using the last age group in the for loop to determine last_time_index_needed. This isn't necessarily the largest index. To include all necessary time points of the past we need the maximum of mean_ExposedToInfectedNoSymptoms[group] + mean_InfectedNoSymptomsToInfectedSymptoms[group] over all groups.